src/PolicyInsights/PolicyInsights/Common/AttestationCmdletBase.cs

```diff
@@ -0,0 +1,118 @@
+
+        /// <summary>
+        /// Gets the root scope of the attestation that is being acted upon.
+        /// </summary>
+        /// <param name="scope">The full scope</param>
+        /// <param name="resourceId">The full resource ID of the attestation resource</param>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="inputObject">The attestation input object</param>
+        protected string GetRootScope(string scope = null, string resourceId = null, string resourceGroupName = null, PSAttestation inputObject = null)
+        {
+            string rootScope = null;
+            if (!string.IsNullOrEmpty(resourceId))
+            {
+                rootScope = ResourceIdHelper.GetRootScope(resourceId: resourceId, fullyQualifiedResourceType: AttestationCmdletBase.AttestationsFullyQualifiedResourceType);
+                if (rootScope == null)
+                {
+                    throw new PSArgumentException(string.Format(CultureInfo.InvariantCulture, Resources.Error_InvalidResourceId, AttestationCmdletBase.AttestationsFullyQualifiedResourceType), paramName: "ResourceId");
+                }
+            }
+            else if (!string.IsNullOrEmpty(scope))
+            {
+                rootScope = scope.TrimEnd('/');
+            }
+            else if (!string.IsNullOrEmpty(resourceGroupName))
+            {
+                rootScope = ResourceIdHelper.GetResourceGroupScope(subscriptionId: this.DefaultContext.Subscription.Id, resourceGroupName: resourceGroupName);
+            }
+            else if (inputObject != null)
+            {
+                rootScope = ResourceIdHelper.GetRootScope(resourceId: inputObject.Id, fullyQualifiedResourceType: AttestationCmdletBase.AttestationsFullyQualifiedResourceType);
+            }
+            else
+            {
+                // Subscription based retrieval is the default, pulls the subscription ID from context
+                rootScope = ResourceIdHelper.GetSubscriptionScope(subscriptionId: this.DefaultContext.Subscription.Id);
+            }
+
+            return rootScope;
+        }
+
+
+        /// <summary>
+        /// Gets the name of the attestation that is being acted upon.
+        /// </summary>
+        /// <param name="name">The provided attestation name</param>
+        /// <param name="resourceId">The full resource ID of the attestation resource</param>
+        /// <param name="inputObject">The attestation input object</param>
+        protected string GetAttestationName(string name = null, string resourceId = null, PSAttestation inputObject = null)
+        {
+            string attestationName = null;
+            if (!string.IsNullOrEmpty(name))
+            {
+                attestationName = name;
+            }
+            else if (!string.IsNullOrEmpty(resourceId))
+            {
+                attestationName = ResourceIdHelper.GetResourceName(resourceId: resourceId);
```

Comment:
> attestationName = ResourceIdHelper.GetResourceName(resourceId: resourceId); [](http://example.com/codeflow?start=15&length=76)

make sure this actually works if the attestation is extending another resource (i.e. [https://learn.microsoft.com/en-us/rest/api/policy/attestations/create-or-update-at-resource?tabs=HTTP](https://learn.microsoft.com/en-us/rest/api/policy/attestations/create-or-update-at-resource?tabs=HTTP))